src/util/mpir_pmi.c

```diff
@@ -415,7 +415,11 @@ static int put_ex(const char *key, const void *buf, int bufsize, int is_local)
         for (i = 0; i < num_segs; i++) {
             char seg_key[50];
             sprintf(seg_key, "%s-seg-%d/%d", key, i + 1, num_segs);
-            encode(segsize, (char *) buf + i * segsize, val);
+            int n = segsize;
```

Comment:
Generally, I'd like to see more in the commit message about what the bugs are and how they are being fixed. Since this is mostly new/untested code, though, I'm not going to put my foot down.

Reply:
I'll add details in the commit message.